eZ/Publish/Core/Persistence/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -826,29 +826,125 @@ public function updateNonTranslatableField(
      *
      * @param mixed $contentId
      * @param mixed $version
-     * @param string[] $translations
+     * @param string[]|null $translations
      *
      * @return array
      */
     public function load($contentId, $version, array $translations = null)
     {
-        $query = $this->queryBuilder->createFindQuery($translations);
-        $query->where(
-            $query->expr->lAnd(
-                $query->expr->eq(
-                    $this->dbHandler->quoteColumn('id', 'ezcontentobject'),
-                    $query->bindValue($contentId)
-                ),
-                $query->expr->eq(
-                    $this->dbHandler->quoteColumn('version', 'ezcontentobject_version'),
-                    $query->bindValue($version)
-                )
+        $results = $this->internalLoadContent([
+            ['id' => $contentId, 'version' => $version, 'languages' => $translations],
+        ]);
+        //if (empty($results)) {
+        // @todo: Should we add a throw here to cleanup and be consistent with other load methods?
```

Comment:
I'd say no, `fetchAll` from `internalLoadContent` will return an empty array `[]`, and this is what should be passed to persistence handler. It should either be a job for persistence handler or even higher layer to decide how to treat empty result. Gateway should just return any result. At least in this case :)

Reply:
ok, the question came up since this is a load single method, and the others throw as far as I saw when I wrote that.